packages/paste-core/components/datagrid/src/DataGrid.tsx

```diff
@@ -0,0 +1,148 @@
+import * as React from 'react';
+import {Box} from '@twilio-paste/box';
+import type {BoxElementProps} from '@twilio-paste/box';
+import {useUID} from '@twilio-paste/uid-library';
+import {useCompositeState, Composite} from '@twilio-paste/reakit-library';
+import {Table} from '@twilio-paste/table';
+import type {TableProps} from '@twilio-paste/table';
+import {DataGridContext} from './DataGridContext';
+
+import {
+  ensureFocus,
+  getActiveElement,
+  getFirstFocusableIn,
+  getClosestCellFrom,
+  getClosestGridCellFromCurrentFocus,
+  isCell,
+  delayedSetFocusable,
+} from './utils';
+
+interface DataGridProps extends TableProps {
+  'aria-label': string;
+  element?: BoxElementProps['element'];
+}
+
+export const DataGrid = React.forwardRef<HTMLTableElement, DataGridProps>(({element = 'DATAGRID', ...props}, ref) => {
+  const dataGridId = `datagrid-${useUID()}`;
+  const lastFocusedElement = React.useRef<Element | null>(null);
+  const compositeState = useCompositeState({unstable_virtual: false});
+  const [actionable, setActionable] = React.useState<boolean>(false);
+
+  const handleMouseDown = React.useCallback(() => {
+    if (!actionable) {
+      setActionable(true);
```

Comment:
Since the state change is being computed using the current state, pretty sure this would need to use a [functional update](https://reactjs.org/docs/hooks-reference.html#functional-updates "React docs - useState - functional updates") like above.

```
setActionable((isActionable) => {
    if (!isActionable) {
        return true; // 
    }
});
```

💡 You can actually simplify this by removing the conditional, since React will only apply the update if the value has changed, [which is mentioned in the API reference](https://reactjs.org/docs/hooks-reference.html#usestate). Check out the terminal logs in this sandbox for an example of that: https://codesandbox.io/s/usestate-functional-and-same-state-update-skipped-iu4kk?file=/src/App.js